### PR TITLE
PS-8447 Document new group_replication options in 8.0.30

### PR DESCRIPTION
**Pull request by patrickbirch**
A  docs/replication/group-replication-flow-control.md A  docs/replication/group-replication-system-variables.md M  mkdocs-base.yml

### CODE REVIEW

File: docs/replication/group-replication-flow-control.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Manage Group Replication flow control
+
+In replication, flow control prevents one member from falling too far behind the cluster and avoid excessive buffering. A cluster is not required to keep members in sync together for replication. The pending transactions in the relay log only increase for the lagging replica.
```
Comment (dutow) [inline docs/replication/group-replication-flow-control.md]: In group replication / for group replication

File: docs/replication/group-replication-flow-control.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Manage Group Replication flow control
+
+In replication, flow control prevents one member from falling too far behind the cluster and avoid excessive buffering. A cluster is not required to keep members in sync together for replication. The pending transactions in the relay log only increase for the lagging replica. 
+
+Flow control helps these members by setting thresholds for transactions to certify and transactions to apply. If the thresholds are exceeded, and during the duration that they are exceeded, flow control adjusts the writer members to the capacity of the delayed member. This action ensures that all members are in sync. Each member sends statistics to the group.
```
Comment (dutow) [inline docs/replication/group-replication-flow-control.md]: "Flow control helps these members by setting thresholds for transactions to certify and transactions to apply." - I'm not sure if I would understand the mechanics based on this sentence.

Flow control sets a threshold on the queue for the transactions waiting in the certification queue, or transactions waiting in the applier queue.

The threshold is for the size of the queue (the number of transactions waiting in the specific queue), not for the number of transactions processed in a given timeframe.

File: docs/replication/group-replication-flow-control.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+* Monitoring the throughput and queue sizes of each member
+* Throttling members to avoid writing beyond the capacity available
+
+The following system variables set flow control behavior in Group Replication:
+
+* [group_replication_flow_control_mode](https://dev.mysql.com/doc/refman/8.0/en/group-replication-options.html#sysvar_group_replication_flow_control_mode)
+* [group_replication_flow_control_certifier_threshold](https://dev.mysql.com/doc/refman/8.0/en/group-replication-options.html#sysvar_group_replication_flow_control_certifier_threshold)
+* [group_replication_flow_control_applier_threshold](https://dev.mysql.com/doc/refman/8.0/en/group-replication-options.html#sysvar_group_replication_flow_control_applier_threshold)
+
+Flow control is enabled and disabled by selecting a value in the group_replication_flow_control_mode variable. Flow control can also be enabled on the certifier or applier level or both and sets the threshold level in seconds.
```
Comment (dutow) [inline docs/replication/group-replication-flow-control.md]: Not in seconds, but in number of transactions waiting in the queue.

File: docs/replication/group-replication-system-variables.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+| Dynamic | Yes |
+| Scope | Global |
+| Type | Integer |
+| Default value | 0 |
+| Maximum Value | 65535  |
+| Unit | seconds |
+
+The value can be changed while Group Replication is running. The change takes effect immediately. Every node in the group can have a different timeout value, but, to avoid unexpected exits, we recommend that all nodes have the same value.
+
+The variable specifies a period of time in seconds before a node is automatically evicted if the node exceeds the [flow control threshold](group-replication-flow-control.md). The default value is 0, which does not change the existing behavior. To set the timeout, change the value with a number higher than zero.
```
Comment (dutow) [inline docs/replication/group-replication-system-variables.md]: "which does not change the existing behavior" - which disables eviction/or the feature?

File: docs/replication/group-replication-system-variables.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+In single-primary mode, a single server ignores a change in the timeout value.
```
Comment (dutow) [inline docs/replication/group-replication-system-variables.md]: "a single server" - no, the primary node. (and it ignores the timeout, not the change)

File: docs/replication/group-replication-system-variables.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## group_replication_flow_control_mode
+
+The variable is in [tech preview](../glossary.md#tech-preview) mode. Before using the variable in production, we recommend that you test restoring production from physical backups in your environment, and also use the alternative backup method for redundancy.
```
Comment (dutow) [inline docs/replication/group-replication-system-variables.md]: The variable isn't in tech preview, our new value (MAJORITY) is.

File: docs/replication/group-replication-flow-control.md (hunk 1 above)
Comment (patrickbirch) [reply docs/replication/group-replication-flow-control.md]: updated.

File: docs/replication/group-replication-flow-control.md (hunk 2 above)
Comment (patrickbirch) [reply docs/replication/group-replication-flow-control.md]: rewritten.

File: docs/replication/group-replication-system-variables.md (hunk 4 above)
Comment (patrickbirch) [reply docs/replication/group-replication-system-variables.md]: updated

File: docs/replication/group-replication-system-variables.md (hunk 6 above)
Comment (patrickbirch) [reply docs/replication/group-replication-system-variables.md]: fixed.

File: docs/replication/group-replication-flow-control.md (hunk 3 above)
Comment (patrickbirch) [reply docs/replication/group-replication-flow-control.md]: removed.

File: docs/replication/group-replication-system-variables.md (hunk 5 above)
Comment (patrickbirch) [reply docs/replication/group-replication-system-variables.md]: Not called primary node

Comment (patrickbirch) [reply docs/replication/group-replication-system-variables.md]: updated.

File: docs/replication/group-replication-flow-control.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# Manage Group Replication flow control
+
+In replication, flow control prevents one member from falling too far behind the cluster and avoids excessive buffering. A cluster is not required to keep members in sync together for replication. The pending transactions in the relay log only increase for the lagging replica. Each member sends statistics to the group.
+
+Flow control sets a threshold on the queue for transactions waiting in the certification queue or the transactions waiting in the applier queue. If the thresholds are exceeded, and during the duration that they are exceeded, flow control adjusts the writer members to the capacity of the delayed member. This action ensures that all members are in sync.
+
+Flow controls work asynchronously and depends on the following:
```
Comment (alina-derkach-oaza) [inline docs/replication/group-replication-flow-control.md]: depends - depend

File: docs/replication/group-replication-system-variables.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+|---|---|
+| Introduced | 8.0.30-22 |
+| Command-line |--group_replication_flow_control_mode |
+| Dynamic | Yes |
+| Scope | Global |
+| Data type | Enumeration |
+| Default value | Quota |
+| Valid values | DISABLED <br> QUOTA <br> MAJORITY |
+
+The variable specifies the mode used flow control.
```
Comment (alina-derkach-oaza) [inline docs/replication/group-replication-system-variables.md]: used flow -> used FOR flow?

File: docs/replication/group-replication-flow-control.md (hunk 7 above)
Comment (patrickbirch) [reply docs/replication/group-replication-flow-control.md]: fixed

File: docs/replication/group-replication-system-variables.md (hunk 8 above)
Comment (patrickbirch) [reply docs/replication/group-replication-system-variables.md]: updated.

**Review by dutow — APPROVED** [review]

**Review by alina-derkach-oaza — APPROVED** [review]
LGTM

**Pull request merged**